pkg/generate/app/cmd/newapp.go

```diff
@@ -901,7 +901,7 @@ func optionallyValidateExposedPorts(config *AppConfig, repositories app.SourceRe
 		return nil
 	}
 
-	if config.Strategy != "docker" {
+	if len(config.Strategy) > 0 && config.Strategy != "docker" {
```

Comment:
there are cases where the strategy is not explicitly specified but new-app detects the build type as dockerfile because a Dockerfile is present in the repo, is this going to handle that properly?